### DISCUSSION TITLE
AppCenter redesign

### DISCUSSION THREAD

**Discussion by pongloongyeat** [elementary/appcenter#1467] [Ideas]
Stumbled upon this on reddit and thought it would be a good idea to post it here. All credits go to @F1L1K.

![](https://preview.redd.it/3jall41wmtd51.png?width=1920&format=png&auto=webp&s=42703fc6e1c0c3930f1e749c66e53eccfd2b519d)

Reddit thread: https://www.reddit.com/r/elementaryos/comments/i01pcn/unofficial_appcenter_redesign_concept/

**Comment by Suzie97** [comment]
![111858321-0ba18d00-895e-11eb-879e-a94c331faa4c](https://user-images.githubusercontent.com/68198116/122256471-b9d0ba80-ceec-11eb-9bf4-ca46737d2ced.png)


This is a redesign by Alecaddd, that I found on his dribbble page.

**Reply by andirsun** [reply]
Better than first mock

**Reply by Alecaddd** [reply]
I'm biased to this one, obviously :sweat_smile:

**Reply by mattbrambilla** [reply]
Love it! 😍

**Comment by elSolus** [comment]
I like F1L1K's design better. And it is a need I guess. AppCenter now feels controlled by others (bcz of lack of the stuff that F1L1K showed)

**Comment by danirabbit** [comment]
I started to prototype a redesign branch here: https://github.com/elementary/appcenter/tree/redesign-prototype

A larger focus on search and less of a focus on switching between these two main view modes. Since we have automatic updates now, we can really start to de-emphasize the updates page.

![Screenshot from 2022-05-02 15 22 18](https://user-images.githubusercontent.com/7277719/166337798-67501439-4a53-4672-9d2c-7d4bc30d6152.png)

One idea here would be that instead of having a view switcher between "Home" and "Installed", we could show the same kind of back button on the "Installed" page as we do other pages. This would free up some space in the header.


![Screenshot from 2022-05-02 15 22 39](https://user-images.githubusercontent.com/7277719/166337794-dc531cbb-d466-4e4a-a64c-ef10b8f385a0.png)

Trying to think about different variations or intermediate steps and what it could look like at small sizes.

![Screenshot from 2022-05-02 15 27 14](https://user-images.githubusercontent.com/7277719/166337790-698faf28-3f98-40c6-8210-a1791692448b.png)
![Screenshot from 2022-05-02 15 27 49](https://user-images.githubusercontent.com/7277719/166337788-998d3842-c4da-4a92-97f5-c7800c01e609.png)

**Reply by ItsJamie9494** [reply]
Personally I dislike having any information except for the app header above screenshots (the last mockup @danrabbit shared has an Outdated tile above the screenshots).

I believe that in order of importance, show the app name and details (authors, summary, download button, icon), functionality (screenshots), then more specific details (description, download size, licenses, runtime issues, etc.)

**Reply by danirabbit** [reply]
@ItsJamie9494 I think it's important to have content warnings above the fold, but we could probably organize things a bit so that there's fewer pixels between the top of the window and screenshots:

![Screenshot from 2022-05-02 16 45 16](https://user-images.githubusercontent.com/7277719/166343589-f2ecce6d-16b3-43eb-bcc8-bf0d549c4e53.png)

**Reply by ItsJamie9494** [reply]
This looks much better in my opinion

**Reply by RMcNeely** [reply]
Is "outdated" a content warning or is it a description based on a determination? I don't see a tag anywhere in the Appstream spec anywhere

**Reply by danirabbit** [reply]
@RMcNeely this warning is specifically about older runtimes. So it’s part of the Flatpak API, not OARS. In this context by “content warning” I mean both warnings from OARS and Flatpak

**Reply by RMcNeely** [reply]
I see.  I think there's an argument to be made that security warnings be presented above the fold with content ratings being ok to be secondary. But this also doesn't sound like a security warning so I'm probably over thinking it.

**Reply by danirabbit** [reply]
@RMcNeely Yeah that's valid! Especially as we add more warnings for Flatpak permissions, it could become pretty crowded. I wouldn't be opposed to splitting things up if there's too much going on in a small space

**Reply by hanaral** [reply]
The original redesign concept is far better than this prototype for a 2 main reasons imo:
A) **the interface hierarchy makes sense**, and fits even the current elementary design guidelines, and,
B) **it isn't a weird mix of the original design, a flat, and adaptive design**.
I think that any sort of interface adaptability can wait until you decide on how the new design will look at its base state
_maybe develop the new granite sidebar replacement with an ability to be hidden/slid open with smaller windows?_

**Reply by danirabbit** [reply]
Just wanted to update anyone following along on this thread that after a ton of refactoring and working I think we're finally in a place to revisit a more aggressive redesign. I have a branch here that puts in view in control of its own header so we can start doing some more interesting things for each view in future branches: https://github.com/elementary/appcenter/pull/2170

**Comment by hanaral** [comment]
I think having a flat headerbar implies a level of functional simplicity that wouldn't logically follow from an app so content-rich that it requires a searchbar (and possibly even search filters in the distant future).
The reason virtually all flat-HB apps elementary and elsewhere have sidebars is due to:
- A surface (in this case the flat HB) that is supposedly the same level as the scrolling UI implies that buttons on it are of the same significance
- But they aren't, they are for overarching actions like updates

So, something more like the original mock-up at the beginning of this thread is most suitable.
Alternatively if you insist that the flat aesthetic is superior to the grey headerbar, then give the flat header a drop-shadow strongly separating it from content.

**Comment by simonsaysthis** [comment]
If possible it would be great if whatever design is chosen it ends up being compatible with Adwaita. Currently the app centre on PopOS does cannot be used properly on Adwaita because several colours are not displayed correctly.

**Reply by danirabbit** [reply]
Pop!_Shop is a significant fork, so you should report any issues you have with it to System76. elementary apps are designed and tested against the elementary stylesheet and while we do our best to work with GNOME for cross-platform compatibility, you should expect apps to break if the stylesheet is changed out from underneath them

**Reply by simonsaysthis** [reply]
Fair enough. Just kinda annoying that Pop devs basically point back at Elementary saying it needs to be requested here.  On the other hand I remember that at some point there was the plan to make the app centre available more universally. I think for this a good first step would be to at least ensure there is compatibility with Adwaita? The issue is that the shop relies on colours which Adwaita and many other themes don’t have.

**Reply by danirabbit** [reply]
We've definitely tried to implement things that System76 has requested upstream, but their fork is several hundred commits behind with another hundred commits added only their fork, so there's not really anything we can do about that. They would have to rebase their fork and stay closer to upstream or maybe just use distro patches instead of a complete fork :shrug: I suppose System76 could carry a stylesheet that provides Granite styles when used with Adwaita or something to that effect, but as far as I know there hasn't been any effort there. We're open to collaboration, but you'd need to talk to System76 about issues you experience in Pop!_OS. We don't have any authority there
